studio/components/interfaces/SQLEditor/RenameQueryModal.tsx

```diff
@@ -56,10 +75,61 @@ const RenameQueryModal = ({ snippet, visible, onCancel, onComplete }: RenameQuer
         {({ isSubmitting }: { isSubmitting: boolean }) => (
           <div className="space-y-4 py-4">
             <Modal.Content>
-              <Input label="Name" id="name" name="name" />
+              <Input
+                label="Name"
+                id="name"
+                name="name"
+                value={nameInput}
+                onChange={(e) => setNameInput(e.target.value)}
+              />
+              <div className="flex w-full justify-end mt-2">
+                {supabaseAIEnabled && !isHipaaSubscription && isAiButtonVisible && (
+                  <Button
+                    type="default"
+                    onClick={async () => {
+                      try {
+                        const { title, description } = await titleSql({
+                          sql: snippet.content.sql,
+                        })
+
+                        setNameInput(title)
```

Comment:
there's a helper method from the Form component `resetForm` that we can leverage on, rather than declaring a separate useState

Reply:
we can leave this as it is for now since either way we're looking to deprecate the use of this Form component
and use the new one that Jonny has done with shadcn